lib/ansible/executor/task_executor.py

```diff
@@ -399,6 +396,33 @@ def _run_loop(self, items):
 
         return results
 
+    def _get_delegated_vars(self, templar, variables):
+        if self._task.delegate_to:
+            delegated_host_name = templar.template(self._task.delegate_to, fail_on_undefined=False)
+            delegated_host = self._variable_manager._inventory.get_host(delegated_host_name)
+            if delegated_host is None:
+                for h in self._variable_manager._inventory.get_hosts(ignore_limits=True, ignore_restrictions=True):
+                    # check if the address matches, or if both the delegated_to host
+                    # and the current host are in the list of localhost aliases
+                    if h.address == delegated_host_name:
+                        delegated_host = h
+                        break
+                else:
+                    delegated_host = Host(name=delegated_host_name)
+
+            variables['ansible_delegated_vars'] = {
+                delegated_host_name: self._variable_manager.get_vars(
+                    play=self._task.get_play(),
+                    host=delegated_host,
+                    task=self._task,
+                    include_delegate_to=False,
+                    include_hostvars=True,
+                )
+            }
+            variables['ansible_delegated_vars'][delegated_host_name]['inventory_hostname'] = variables.get('inventory_hostname')
+            # At the point this is executed, `self._task` is a copy referred to by `tmp_task` in `_execute`
```

Comment:
I suspect the comment informs that it is safe to overwrite `Task`'s attribute because it is a copy. But it only makes sense if `_execute` is called in a loop from `_run_loop`, right?

Reply:
Yes, I've clarified the comment. The line below this comment will only happen in a loop context, but regardless it's still a copy at this point.